jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.cpp

```diff
@@ -733,25 +734,66 @@ ecma_date_make_day (ecma_number_t year, /**< year value */
     return ecma_number_make_nan ();
   }
 
+  /* 2., 3., 4. */
   ecma_number_t y = ecma_number_trunc (year);
   ecma_number_t m = ecma_number_trunc (month);
   ecma_number_t dt = ecma_number_trunc (date);
+  /* 5. */
   ecma_number_t ym = y + (ecma_number_t) floor (m / 12);
+  /* 6. */
   ecma_number_t mn = (ecma_number_t) fmod (m, 12);
   mn = (mn < 0) ? 12 + mn : mn;
+
+  /* 7. */
   ecma_number_t time = ecma_date_time_from_year (ym);
+  ecma_number_t step = floor ((ecma_number_t) (365) / 2);
+  ecma_number_t delta = step;
+
+  /**
+   * The algorithm below searches the following date: ym-mn-1
+   * To find this time it starts from the beggining of the year (ym)
+   * then increase it by ECMA_DATE_MS_PER_DAY until it reaches the
+   * right date.
+   */
+  if (ecma_date_year_from_time (time) == ym)
```

Comment:
Hi, I have made a quick test with every date in a year.
The binary search always takes 9 iterations and in the worst case it needs 4 iterations in the day by day phase while in the best case it skips it. The whole test took 0.458s.
The old algorithm is far worse because it needs 335 (december) iterations at the worst case but 0 in the best case (january). The test took 0.574s.
We could use lookup table for the months to get the days but I think we won't get any measurable performance gain in normal usage with this function.